app/helpers/map_helper.rb

```diff
@@ -22,8 +22,10 @@ def obfuscated_coordinates(init_lat, init_lng)
     sin_lat = Math.sin(lat)
     cos_lat = Math.cos(lat)
 
-    randomize_bearing = rand
-    randomize_distance = rand
+    cache_key = "#{listing_id}#{init_lat}#{init_lng}".gsub(/[^0-9]/, '').reverse.to_i
+
+    randomize_bearing = Random.new(cache_key).rand
+    randomize_distance = Random.new(cache_key).rand
```

Comment:
No, this will not work.

Example:

s = Random.new(12345)
=> #<Random:0x0000000007b73e30>

s.rand
=> 0.9296160928171479

s.rand
=> 0.3163755545817859

s.rand
=> 0.18391881167709445

**But**

>> >> Random.new(12345).rand
=> 0.9296160928171479

>> >> Random.new(12345).rand
=> 0.9296160928171479

Reply:
@Charly3X this works fine - it's important that the values for `randomize_bearing` are the same each time. Same for `randomize_distance`. However, it's not important that `randomize_bearing == randomize_distance`. Or am I missing something?

See - you get the same sequence of values each time you set the seed.

```ruby
$ irb
2.3.4 :001 > r = Random.new(12345)
 => #<Random:0x007f903505cea0>
2.3.4 :002 > randomize_bearing = r.rand
 => 0.9296160928171479
2.3.4 :003 > randomize_distance = r.rand
 => 0.3163755545817859
2.3.4 :004 > r = Random.new(12345)
 => #<Random:0x007f90358566b0>
2.3.4 :005 > randomize_bearing = r.rand
 => 0.9296160928171479
2.3.4 :006 > randomize_distance = r.rand
 => 0.3163755545817859
```

Reply:
I agree if this is randomize_bearing == randomize_distance not important